qutip/bloch.py

```diff
@@ -394,6 +403,97 @@ def add_annotation(self, state_or_vector, text, **kwargs):
                                  'text': text,
                                  'opts': kwargs})
 
+    def add_arc(self, start, end, fmt = "b", steps = None, **kwargs):
+        """Adds an arc between two points on a sphere. The arc is set to be
+        blue solid curve by default.
+        
+        Parameters
+        ----------
+        point1 : array_like
+            Array with cartesian coordinates of the first point.
+        point2 : array_like
+            Array with cartesian coordinates of the second point.
+        """
+        if isinstance(start, Qobj):
+            start = [start]
+            for st in start:
+                pt1 = [expect(sigmax(), st),
+                           expect(sigmay(), st),
+                           expect(sigmaz(), st)]
+        else:
+            pt1 = start
```

Comment:
I believe we are assuming the input can only be a single `Qobj`, or an `array_like`, dont we? If that is the case the following code will do the job:

```suggestion
        if isinstance(start, Qobj):
                start = [expect(sigmax(), start),
                            expect(sigmay(), start),
                            expect(sigmaz(), start)]
```
A similar change would be will be needed for `end`.

Reply:
Ah, I see. I will make the appropriate changes.